packages/rulesets/src/spectral/functions/security-definitions-structure.ts

```diff
@@ -0,0 +1,44 @@
+// Checks for the presence and existence of the security definition
+
+import { isSchemaEqual } from "./utils";
+
+export const securityDefinitionsStructure = (swagger: any, _opts: any) => {
+  if (swagger === "" || typeof swagger !== "object") {
+    return [];
+  }
+  if (!Object.keys(swagger).includes("securityDefinitions")) {
+    return [];
+  }
+  const errors: any = [];
+  const securityDefinitionsModule = {
+    azure_auth: {
+      type: "oauth2",
+      authorizationUrl: "https://login.microsoftonline.com/common/oauth2/authorize",
+      flow: "implicit",
+      description: "Azure Active Directory OAuth2 Flow",
+      scopes: {
+        user_impersonation: "impersonate your user account",
+      },
+    },
+  };
+  const securityDefinition = swagger.securityDefinitions;
+  const securityDefinitionClone = JSON.parse(JSON.stringify(securityDefinition));
+  if (
+    securityDefinitionClone.azure_auth?.description &&
+    securityDefinitionClone.azure_auth.description !== ""
+  ) {
+    securityDefinitionClone.azure_auth.description = "Azure Active Directory OAuth2 Flow";
```

Comment:
only need to the description and user_impersonation  is not empty

Reply:
it's a little confusion here, you can just compare the 'description' and other fields directly, don't need to clone the securityDefinition

Reply:
I mean you don't need to use isSchemaEqual , just compare the field directly